### PR TITLE
Expose lane icon drawables

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

The drawable's we have for lane icons is useful. When adding lane guidance to [android-auto](https://github.com/mapbox/mapbox-navigation-android-examples/tree/main/android-auto) we need these drawables to create a bitmap. 

### Changelog

```
<changelog>Expose lane icon drawables with `MapboxLaneIconsApi`</changelog>
```

### Screenshots or Gifs

four_lanes_right_left_straight_right

Here is some sample code of how we're generating the bitmaps for lane guidance in android auto

``` kotlin
val carLaneBitmap = Bitmap.createBitmap(widthPx, heightPx, Bitmap.Config.ARGB_8888)
carLaneBitmap.eraseColor(themeProvider.backgroundColor())

val canvas = Canvas(carLaneBitmap)
carLaneIcons.forEach { carLaneIcon ->
    val vectorDrawable = VectorDrawableCompat.create(
        context.resources,
        carLaneIcon.laneIcon,
        themeProvider.theme(ContextThemeWrapper(context, R.style.CarLaneRestrictedTheme).theme)
    )!!
    val iconBounds = calculateBounds(carLaneIcon)
    vectorDrawable.bounds = iconBounds
    vectorDrawable.draw(canvas)
}

return CarIcon.Builder(
    IconCompat.createWithBitmap(carLaneBitmap)
).build()
```

And then you create your own styles for lane guidance
``` xml
<resources>
    <style name="CarLaneRestrictedTheme" parent="YourAppTheme">
        <item name="maneuverTurnIconColor">#D50000</item>
        <item name="maneuverTurnIconShadowColor">#767676</item>
    </style>
</resources>
```

### CODE REVIEW

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxLaneIconsApi.kt [hunk 1]
```diff
@@ -48,7 +54,12 @@ internal class LaneIconHelper {
             R.drawable.mapbox_ic_lane_sharp_right_straight_straight_only,
     )
 
-    fun retrieveLaneToDraw(laneIndicator: LaneIndicator, activeDirection: String?): Int? {
+    /**
+     * Given a [LaneIndicator] ane [activeDirection] which can be retrieved using
+     * the [MapboxManeuverApi], get a drawable resource that represents the lane guidance.
+     */
+    @DrawableRes
+    fun drawable(laneIndicator: LaneIndicator, activeDirection: String?): Int? {
```
Comment (kmadsen) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxLaneIconsApi.kt:62]: wondering if the public api should actually be exposing a `Maneuver`

``` kotlin
fun drawable(maneuver: Maneuver): Int? {
    return ifNonNull(maneuver.laneGuidance, maneuver.laneGuidance?.activeDirection) {
        laneGuidance, activeDirection ->
...
```

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/LaneIconProcessor.kt [hunk 2]
```diff
@@ -48,7 +48,8 @@ internal class LaneIconHelper {
             R.drawable.mapbox_ic_lane_sharp_right_straight_straight_only,
     )
 
-    fun retrieveLaneToDraw(laneIndicator: LaneIndicator, activeDirection: String?): Int? {
+    @DrawableRes
+    internal fun drawable(laneIndicator: LaneIndicator, activeDirection: String?): Int? {
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/LaneIconProcessor.kt:52]: We can remove `internal` to maintain consistency with other `Processors`. 
Also let's change the function name to `getDrawableFrom` as `drawable` is not intuitive

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxLaneIconsApi.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.navigation.ui.maneuver.LaneIconProcessor
+import com.mapbox.navigation.ui.maneuver.model.LaneIcon
+import com.mapbox.navigation.ui.maneuver.model.LaneIconError
+import com.mapbox.navigation.ui.maneuver.model.LaneIndicator
+import com.mapbox.navigation.utils.internal.ifNonNull
+
+/**
+ * MapboxLaneIconsApi allows you to request the lane icon used to create lane guidance.
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxLaneIconsApi.kt:12]:
```suggestion
 * MapboxLaneIconsApi allows you to request the lane icon from [LaneIndicator] in the form of vector drawables which can then be styled and rendered using [MapboxLaneGuidance] or your own custom view.
```

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxLaneIconsApi.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * This allows you to apply your own styles to Mapbox vector drawables.
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxLaneIconsApi.kt:13]: Lets remove this.

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/model/LaneIcon.kt [hunk 5]
```diff
@@ -0,0 +1,45 @@
+package com.mapbox.navigation.ui.maneuver.model
+
+import androidx.annotation.DrawableRes
+
+/**
+ * Data structure that holds a styleable drawable resource.
+ *
+ * Within your style, specify the attributes to color the resource.
+ *   maneuverTurnIconColor. For the active lane
+ *   maneuverTurnIconShadowColor. For the not active lane
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/model/LaneIcon.kt:10]:
```suggestion
 *   maneuverTurnIconShadowColor. For inactive lanes
```

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/model/LaneIcon.kt [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package com.mapbox.navigation.ui.maneuver.model
+
+import androidx.annotation.DrawableRes
+
+/**
+ * Data structure that holds a styleable drawable resource.
+ *
+ * Within your style, specify the attributes to color the resource.
+ *   maneuverTurnIconColor. For the active lane
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/model/LaneIcon.kt:9]:
```suggestion
 *   maneuverTurnIconColor. For active lanes
```

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxLaneGuidance.kt [hunk 7]
```diff
@@ -5,14 +5,16 @@ import android.util.AttributeSet
...
+import com.mapbox.navigation.ui.maneuver.api.MapboxLaneIconsApi
 import com.mapbox.navigation.ui.maneuver.model.LaneIndicator
-import com.mapbox.navigation.ui.utils.internal.ifNonNull
+import com.mapbox.navigation.utils.internal.LoggerProvider
 
 /**
  * Default Lane Guidance View that renders the maneuver icons onto [MapboxManeuverView].
  * It can be directly used in any other layout.
- * @property laneIconHelper LaneIconHelper
+ * @property laneIconsApi LaneIconHelper
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxLaneGuidance.kt:17]:
```suggestion
 * @property laneIconsApi MapboxLaneIconsApi
```

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxLaneGuidance.kt [hunk 8]
```diff
@@ -34,11 +36,11 @@ class MapboxLaneGuidance @JvmOverloads constructor(
         activeDirection: String?,
         wrapper: ContextThemeWrapper
     ) {
-        val laneIcon = laneIconHelper.retrieveLaneToDraw(laneIndicator, activeDirection)
-        ifNonNull(laneIcon) { icon ->
+        val laneIcon = laneIconsApi.laneIcon(laneIndicator, activeDirection)
+        laneIcon.onValue { icon ->
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxLaneGuidance.kt:40]:
```suggestion
        laneIcon.fold(
            { error ->
                LoggerProvider.logger.i(Tag("MapboxLaneGuidance"), Message(error.errorMessage))
            },
            { icon ->
                val drawable = VectorDrawableCompat.create(
                    context.resources,
                    icon.drawableResId,
                    wrapper.theme
                )
                setImageDrawable(drawable)
                alpha = if (laneIndicator.isActive) {
                    1.0f
                } else {
                    0.5f
                }
            }
        )```

**Review by abhishek1508 — COMMENTED** [review]
LGTM. Left some comments

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/LaneIconProcessor.kt (hunk 2 above)
Comment (kmadsen) [reply libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/LaneIconProcessor.kt:52]: agree 👍  
IMO prefer `map` over `get` for these `mapDrawableFrom`. will do get for now to be consistent

**Review by abhishek1508 — APPROVED** [review]
🚀

**Comment by kmadsen** [comment]
https://console.firebase.google.com/u/0/project/mapbox-navigation-android/testlab/histories/bh.54eb2dd57e959221/matrices/6084049207900269213/executions/bs.9d8bc1a89dee01d8

`reroute_completes` failing and is existing https://github.com/mapbox/mapbox-navigation-android/issues/4429

**Pull request merged**